src/app/vault/password-generator.component.html

```diff
@@ -27,12 +27,16 @@
                         </button>
                     </div>
                     <div class="box-content condensed" [hidden]="!showOptions">
-                        <div class="box-content-row" appBoxRow>
-                            <label for="type" class="sr-only">{{'type' | i18n}}</label>
-                            <select id="type" [(ngModel)]="options.type" (change)="saveOptions()">
-                                <option value="password">{{'password' | i18n}}</option>
-                                <option value="passphrase">{{'passphrase' | i18n}}</option>
-                            </select>
+                        <div class="box-content-row box-content-row-radio">
+                            <div class="radio-group text-default" appBoxRow name="PassTypeOptions"
+                                *ngFor="let o of passTypeOptions">
+                                <input type="radio" class="radio" [(ngModel)]="options.type" name="Type_{{o.value}}"
+                                    id="type_{{o.value}}" [value]="o.value" (change)="saveOptions()"
+                                    [checked]="options.type === o.value">
+                                <label class="unstyled" for="type_{{o.value}}">
+                                    {{o.name | i18n}}
```

Comment:
Ah, I didn't realise you'd done the internationalization here to begin with. Anyway, now that we've got the call to `i18nService.t` in the base class (in jslib), we can remove the `i18n` pipe here. So this line will just be: `{{o.name}}`